### PR TITLE
Replace timer with monotonic time in gproc_pool

### PR DESCRIPTION
**Pull request by leoliu**
I have often observed stray timeout messages even after modifying clear_wait/1 to flush out these messages like so:

clear_wait(nowait) ->
    ok;
clear_wait({busy_wait, Ref}) ->
    erlang:cancel_timer(Ref),
    receive {timeout, Ref, _} -> ok
    after 0 -> ok
    end.

The documentation of erlang:cancel_timer/2 has this to say:

  ...Even if the timer had expired, it does not tell you if the time-out
  message has arrived at its destination yet.

That means getting rid of unwanted timeout messages is racy and expensive. In this change monotonic time is used to detect timeout which avoid this problem.

### CODE REVIEW

**Review by uwiger — COMMENTED** [review]
Yes, very nice.

**Comment by uwiger** [comment]
Out of curiosity, I checked the timeline. The gproc_pool code was written a year before the new time API. That is at least some excuse for not thinking of this right away. ;-)

File: src/gproc_pool.erl [hunk 1]
```diff
@@ -557,32 +556,20 @@ execute_claim(F, K, Pid) ->
 setup_wait(nowait, _) ->
     nowait;
 setup_wait({busy_wait, MS}, Pool) ->
```
Comment (uwiger) [inline src/gproc_pool.erl:558]:
```suggestion
setup_wait({busy_wait, MS}, _Pool) ->
```

**Review by uwiger — COMMENTED** [review]
The change introduced an unused variable warning, though

File: src/gproc_pool.erl (hunk 1 above)
Comment (leoliu) [reply src/gproc_pool.erl]: @uwiger thanks for the review. Now fixed and force-pushed.

**Review by uwiger — APPROVED** [review]

**Comment by uwiger** [comment]
Thanks. Merged.

**Pull request merged**